webapp/src/components/preview-movies/style/row/TopRow.tsx

```diff
@@ -1,54 +1,119 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { Production } from '../../../../api/types/MovieDbTypes';
 import RowLoading from '../../../general/loading/RowLoading';
-import '../../../../../assets/scss/components/row.scss';
-import '../../../../../assets/scss/components/arrow.scss';
-import '../../../../../assets/scss/components/top-row.scss';
-import Player from '../../../general/streaming/movie/Player';
-import Arrow from './Arrow';
-import Poster from '../image/Poster';
+import '../../../../../assets/scss/components/style/row/row.scss';
+import '../../../../../assets/scss/components/style/arrow/arrow.scss';
+import '../../../../../assets/scss/components/style/row/top-row.scss';
+import Arrow from '../arrow/Arrow';
+import Poster from '../poster/Poster';
 
 type Props = {
   title?: string,
   movieList: Production[],
   isDataLoading?: boolean
 };
 
+const SLIDER_TIMING: number = 8000;
+
 function TopRow({ movieList, isDataLoading }: Props) {
-  function hundleClickArrowRight() {
-    const slider = document.getElementById('top_posters');
-    slider.scrollLeft += window.innerWidth;
+  const slider = useRef<HTMLDivElement>(null);
+
+  const [currentPoster, setCurrentPoster] = useState(0);
+
+  const sliderInterval = useRef<NodeJS.Timeout>();
+
+  function createSliderInterval() {
+    if (sliderInterval.current) {
+      stopSliderInterval();
+    }
+
+    sliderInterval.current = setInterval(() => goNextPoster(), SLIDER_TIMING);
   }
 
-  function hundleClickArrowLeft() {
-    const slider = document.getElementById('top_posters');
-    slider.scrollLeft -= window.innerWidth;
+  function stopSliderInterval() {
+    if (!sliderInterval?.current) {
+      return;
+    }
+
+    return clearInterval(sliderInterval.current);
   }
 
+  function goNextPoster() {
+    setCurrentPoster((prevPoster) => {
+      if (!slider.current) {
+        return prevPoster;
+      }
+
+      stopSliderInterval();
+
+      if (prevPoster + 1 >= movieList.length) {
+        slider.current.scrollLeft = 0;
+        return 0;
+      }
+      slider.current.scrollLeft += window.innerWidth;
+
+      createSliderInterval();
+
+      return prevPoster + 1;
+    });
+  }
+
+  function goPreviousPoster() {
+    if (!slider.current) {
+      return;
+    }
+
+    stopSliderInterval();
+
+    setCurrentPoster(currentPoster - 1);
+    slider.current.scrollLeft -= window.innerWidth;
+
+    createSliderInterval();
+  }
+
+  useEffect(() => {
+    createSliderInterval();
+
+    // Nettoyer l'intervalle lorsque le composant est démonté pour éviter les fuites de mémoire
+    stopSliderInterval();
```

Comment:
il faut la return cette fonction, pour qu'elle soit éxecuté au unmount

```suggestion
    return stopSliderInterval;
```